test/plugin_functional/test_suites/core_plugins/application_deep_links.ts

```diff
@@ -42,65 +46,64 @@ export default function ({ getService, getPageObjects }: PluginFunctionalProvide
     });
   };
 
-  const navigateToAppLinks = async (subject: string) => {
-    if (!(await testSubjects.exists(subject))) {
-      log.debug(`side nav in app not in DOM`);
-    }
-    await testSubjects.click(subject);
+  const loadingScreenNotShown = async () =>
+    expect(await testSubjects.exists('kbnLoadingMessage')).to.be(false);
+
+  const checkAppVisible = async (app: string) => {
+    const appContainer = `dlApp${app}`;
+    await testSubjects.existOrFail(appContainer);
   };
 
-  // FLAKY: https://github.com/elastic/kibana/issues/166893
-  describe.skip('application deep links navigation', function describeDeepLinksTests() {
+  describe('application deep links navigation', function describeDeepLinksTests() {
     before(async () => {
       await esArchiver.emptyKibanaIndex();
       await PageObjects.common.navigateToApp('dl');
     });
 
     it('should start on home page', async () => {
-      await testSubjects.existOrFail('dlAppHome');
+      await checkAppVisible('Home');
     });
 
     it('should navigate to page A when navlink is clicked', async () => {
-      await navigateToAppLinks('dlNavPageA');
+      await clickAppLink('PageA');
       await waitForUrlToBe('/app/dl/page-a');
       await loadingScreenNotShown();
-      await testSubjects.existOrFail('dlNavPageA');
+      await checkAppVisible('PageA');
     });
 
     it('should be able to use the back button to navigate back to previous deep link', async () => {
       await browser.goBack();
       await waitForUrlToBe('/app/dl/home');
       await loadingScreenNotShown();
-      await testSubjects.existOrFail('dlNavHome');
+      await checkAppVisible('Home');
     });
 
     it('should navigate to nested page B when navlink is clicked', async () => {
-      await navigateToAppLinks('dlNavDeepPageB');
+      await clickAppLink('DeepPageB');
       await waitForUrlToBe('/app/dl/page-b');
       await loadingScreenNotShown();
-      await testSubjects.existOrFail('dlNavDeepPageB');
+      await checkAppVisible('PageB');
     });
 
     it('should navigate to Home when navlink is clicked inside the defined category group', async () => {
-      await appsMenu.clickLink('DL Home', { category: 'securitySolution' });
```

Comment:
This is the previous/old navigation logic that should have been removed.
I believe this is the source of the new / reoccurring flakiness.